### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by njm222**
Added oAuth authentication with github & google

@SaraVieira You will need to turn on google as a provider & restrict the `scores` routes to authenticated only

### CODE REVIEW

File: src/ui/Finished.jsx [hunk 1]
```diff
@@ -2,60 +2,57 @@ import { useStore, mutation } from '../store'
...
-  const [savedName, setSavedName] = useState(window.localStorage.getItem(LOCAL_STORAGE_KEY))
+
+  const [name, setName] = useState(window.localStorage.getItem(LOCAL_STORAGE_KEY))
   const [leaderBoard, setLeaderBoard] = useState(null)
   const [position, setPosition] = useState(null)
 
   const sendTime = async () => {
-    const newTime = await insertTime({ time: finished, name: savedName })
+    window.localStorage.setItem(LOCAL_STORAGE_KEY, name)
+    const newTime = await insertTime({ time: finished, name: name })
```
Comment (njm222) [inline src/ui/Finished.jsx:19]:
```suggestion
    const newTime = await insertTime({ time: finished, name })
```

File: src/ui/Auth.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { authenticateUser } from '../utils/data/auth'
+import googleLogo from '../icons/google-logo.png'
+import githubLogo from '../icons/github-logo.png'
+
+const providers = [
+  {
+    provider: 'goggle',
```
Comment (bjornstar) [inline src/ui/Auth.jsx:7]:
```suggestion
    provider: 'google',
```

**Comment by drcmda** [comment]
are we finished here @njm222 ? anything that needs to done before we merge?

**Comment by njm222** [comment]
Yes it is ready to go! I think Sara still needs to turn on google as a provider and restrict the routes but it can be merged without those changes

File: src/data.js [hunk 3]
```diff
@@ -23,3 +23,16 @@ export const insertScore = ({ time, name }) =>
     .from('scores')
     .insert({ time, name })
     .then(({ data }) => data)
+
+export const setupSession = (set) => {
+  const supaSession = client.auth.session()
+  set({ session: supaSession })
```
Comment (njm222) [inline src/data.js:29]:
```suggestion
  set({ session: client.auth.session() })
```

**Comment by njm222** [comment]
Good to go! (merged in main and fixed a minor issue)

**Pull request merged**